apps/guardian/src/pages/Actions/SetupActions.tsx

```diff
@@ -534,22 +253,33 @@ function Invalid() {
  */
 export default function SetupActions() {
     const electionStep = useAtomValue(electionStepAtom);
-    const { guardians } = useAtomValue(guardiansStateAtom);
 
-    if (electionStep?.phase !== ElectionPhase.Setup || guardians === undefined) {
+    if (electionStep?.phase !== ElectionPhase.Setup) {
         return null;
     }
 
     return (
-        <div className="text-center">
+        <>
             {electionStep.step === SetupStep.GenerateKey && <GenerateGuardianKey />}
-            {electionStep.step === SetupStep.AwaitPeerKeys && <AwaitPeerKeys guardians={guardians} />}
+            {electionStep.step === SetupStep.AwaitPeerKeys && (
+                <AwaitPeers predicate={(g) => g.hasPublicKey}>
+                    Waiting for other guardians to register their keys
```

Comment:
I'm guessing people will have opinions about these phrasings.

Perhaps it'd be useful to ask at the meeting about how this should be phrased.

Reply:
I guess we'll see 😄